core/lib/vm_interface/src/types/outputs/execution_result.rs

```diff
@@ -118,6 +118,7 @@ pub struct VmExecutionResultAndLogs {
     pub logs: VmExecutionLogs,
     pub statistics: VmExecutionStatistics,
     pub refunds: Refunds,
+    pub new_known_factory_deps: Option<Vec<(H256, Vec<u8>)>>,
```

Comment:
I think I understand the semantics of the field now: it includes dynamically deployed bytecodes (i.e., ones not mentioned in factory deps of a transaction). If this is the case, this raises further questions:

- Why is this field optional? If old VMs don't support dynamically deployed bytecodes, then I'd argue it would be caller's responsibility to not provide them with EVM simulator-enabled env.
- I'd argue that a `HashMap<H256, Vec<u8>>` would represent field semantics better. IIUC, bytecodes *can* be deduplicated by hash.